runtime/lua/vim/fs.lua

```diff
@@ -367,4 +367,33 @@ function M.normalize(path, opts)
   return (path:gsub('(.)/$', '%1'))
 end
 
+--- Check if a path is contained within another path.
+---@param container string The path being checked to see if it is the parent
+---@param path string The path being checked to see if it is contained within the parent
+---@return boolean
+function M.path_contains(container, path)
+  -- Normalize both paths so we don't have to deal with different path separators
+  container = M.normalize(vim.fn.fnamemodify(container, ':p'))
+  path = M.normalize(vim.fn.fnamemodify(path, ':p'))
```

Comment:
`libuv` has `fs_realpath`, it may be better to use libuv's functions instead of `vim.fn.*` to allow the method to be used in fast event context and leave the door open for users to use them in threads

Reply:
If we don't use libuv in a fairly recent API, there's a reason for it. Vim's function are (sadly, for libuv), more robust and platform-independent especially around edge cases.